contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -63,24 +63,35 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     bool isActive;
   }
 
-  IERC20 private immutable LINK_TOKEN;
+  bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+  bytes32 public constant EXECUTOR_ROLE = keccak256("EXECUTOR_ROLE");
+  uint96 private constant DEFAULT_TOP_UP_AMOUNT_JULES = 9000000000000000000;
+  uint96 private constant DEFAULT_MIN_BALANCE_JULES = 1000000000000000000;
+  IERC20 private immutable i_linkToken;
+
   uint256 private s_minWaitPeriodSeconds;
   uint16 private s_maxPerform;
   uint16 private s_maxCheck;
   uint8 private s_upkeepInterval;
   address[] private s_watchList;
   mapping(address targetAddress => MonitoredAddress targetProperties) internal s_targets;
+  mapping(uint64 dstChainSelector => address onRamp) internal s_onRampAddresses;
 
   /// @param linkTokenAddress the LINK token address
   constructor(
+    address admin,
     address linkTokenAddress,
     uint256 minWaitPeriodSeconds,
     uint16 maxPerform,
     uint16 maxCheck,
     uint8 upkeepInterval
-  ) ConfirmedOwner(msg.sender) {
+  ) {
     if (linkTokenAddress == address(0)) revert InvalidLinkTokenAddress(linkTokenAddress);
-    LINK_TOKEN = IERC20(linkTokenAddress);
+    _setRoleAdmin(ADMIN_ROLE, ADMIN_ROLE);
+    _setRoleAdmin(EXECUTOR_ROLE, ADMIN_ROLE);
+    _grantRole(ADMIN_ROLE, admin);
+    _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
```

Comment:
iiuc, we are overriding the default admin role for `ADMIN_ROLE` and `EXECUTOR_ROLE`, meaning the the `DEFAULT_ADMIN_ROLE` is now useless and has no privileges. Therefore, I think we can probably avoid setting any users to that role.